constructor/osxpkg.py

```diff
@@ -188,6 +189,27 @@ def modify_xml(xml_path, info):
         )
         root.append(readme)
 
+    # -- __osx virtual package checks -- #
+    osx_versions = {}
+    for spec in info.get("virtual_specs", ()):
+        spec = MatchSpec(spec)
+        if spec.name != "__osx":
+            continue
+        if not spec.version:
+            continue
+        operator = spec.version.operator_func.__name__
+        if operator == "ge":
+            osx_versions["min"] = str(spec.version.matcher_vo)
+        elif operator == "le":
+            osx_versions["max"] = str(spec.version.matcher_vo)
```

Comment:
This part makes a lot of assumptions about the `MatchSpec` that aren't documented. For example, version ranges or `>` or `<` are not recognized. This should either be implemented or documented.

Reply:
Agreed. Not sure how to deal with those (we'd need some arithmetics). This is just a "fast-track" for common cases (`>=`, `<=`). The regular conda solver check will run eventually if this one is not included, though. Maybe it's just a matter of documenting this in the `virtual_specs` key? Something like:

> In PKG installers, `__pkg>=x.y` and `__pkg<=x.y` can be checked natively (without the conda solver being involved) before the installation actually begins. Other specs will be checked later in the process.

Reply:
Added to docs

Reply:
I think they should at least error out. You could also cover the `>=,<=` case this way:

```suggestion
        versions = spec.version.tup if "," in spec.version.spec_str else (spec.version,)
        for version in versions:
            operator = version.operator_func.__name__
            if operator == "ge":
                osx_versions["min"] = str(version.matcher_vo)
            elif operator == "le":
                osx_versions["max"] = str(version.matcher_vo)
            else:
                raise ValueError(f"Invalid operator for {version}. Must be one of `<=` or `>=`.")
```

Reply:
Added this but had to tune it a bit. It's not `max` but `before` (yes, I don't know why). And the docs say it's a `<` for `before`, so I'll only allow that. Technically we can also support OR via `|` using several `os-version` keys but I don't think that's super useful so I won't add it and instead reject it explicitly.